core/src/style/material.cpp

```diff
@@ -158,56 +158,56 @@ void Material::injectOnProgram(ShaderProgram& _shader ) {
 void Material::setupProgram(ShaderProgram& _shader) {
 
     if (m_bEmission) {
-        _shader.setUniformf("u_material.emission", m_emission);
+        _shader.setUniformf(m_uEmission, m_emission);
```

Comment:
@hjanetzek This is working properly for lights since there is a specific `LightUniform` which targets the shader it gets injected into. 

But now since `UniformLocation` now owns the location, if a material is injected to several styles (e.g. to different `ShaderProgram`) the location is now inconsistent.

Can you verify that?